openquake/hazardlib/contexts.py

```diff
@@ -859,10 +860,9 @@ def gen_contexts(self, rups_sites, src_id):
                     rctx.src_id = src_id
                     if src_id >= 0:  # classical calculation
                         rctx.rup_id = rup.rup_id
-                        if self.fewsites:
-                            c = rup.surface.get_closest_points(sites.complete)
-                            rctx.clon = c.lons[rctx.sids]
-                            rctx.clat = c.lats[rctx.sids]
+                        c = rup.surface.get_closest_points(sites.complete)
```

Comment:
Now  `rup.surface.get_closest_points(sites.complete)` is called always while before if was not called in large calculations, so you added a performance penalty without need.

Reply:
I have removed this modification (unit tests and qa tests still pass as before